proxy/http/HttpTransactCache.cc

```diff
@@ -949,17 +949,17 @@ HttpTransactCache::calculate_quality_of_accept_encoding_match(MIMEField *accept_
   // still okay, but otherwise, this is just not a match at all.         //
   /////////////////////////////////////////////////////////////////////////
   if ((q == -1.0) && is_identity_encoding) {
-    if (match_gzip(accept_field) == GZIP) {
-      if (match_gzip(cached_accept_field) == GZIP) {
-        return (float)1.0;
+    if (match_content_encoding(accept_field, "gzip")) {
+      if (match_content_encoding(cached_accept_field, "gzip")) {
+        return float(1);
       } else {
         // always try to fetch GZIP content if we have not tried sending AE before
-        return (float)-1.0;
+        return float(-1);
       }
-    } else if (cached_accept_field && match_gzip(cached_accept_field) != GZIP) {
-      return (float)0.001;
+    } else if (!cached_accept_field && match_content_encoding(cached_accept_field, "gzip")) {
```

Comment:
What is this checking? If `!cached_accept_field` won't the call to `match_content_encoding` always fail? Why check it?

Reply:
Typo, should be 
{noformat}
} else if (cached_accept_field and !match_content_encoding(cached_accept_field, "gzip")) {
{noformat}

Reply:
OK fixed and squashed.